pkg/workflows/upgrade_test.go

```diff
@@ -518,3 +523,69 @@ func TestUpgradeWorkloadRunSuccess(t *testing.T) {
 		t.Fatalf("Upgrade.Run() err = %v, want err = nil", err)
 	}
 }
+
+func TestUpgradeRunFailedUpgradeWithCheckpoint(t *testing.T) {
+	test := newUpgradeSelfManagedClusterTest(t)
+	test.expectSetup(test.workloadCluster)
+	test.expectPreflightValidationsToPass()
+	test.expectUpdateSecrets(test.workloadCluster)
+	test.expectEnsureEtcdCAPIComponentsExistTask(test.workloadCluster)
+	test.expectUpgradeCoreComponents(test.workloadCluster, test.workloadCluster)
+	test.expectProviderNoUpgradeNeeded(test.workloadCluster)
+	test.expectVerifyClusterSpecChanged(test.workloadCluster)
+	test.expectPauseEKSAControllerReconcile(test.workloadCluster)
+	test.expectPauseGitOpsKustomization(test.workloadCluster)
+	test.expectCreateBootstrap()
+	test.expectMoveManagementToBootstrap()
+	test.expectUpgradeWorkloadToReturn(test.bootstrapCluster, test.workloadCluster, errors.New("failed upgrading"))
+	test.expectSaveLogs(test.workloadCluster)
+
+	err := test.run()
+	if err == nil {
+		t.Fatal("Upgrade.Run() err = nil, want err not nil")
+	}
+	features.ClearCache()
```

Comment:
why are we calling this here?

Reply:
I believe you told me about how we need to clear the cache sometimes for unit tests with env vars. I was running into odd behavior without clearing the cache for this unit test. After I added this, it began to work as expected. We do this in other unit tests with gated features as well.